src/main/java/com/google/devtools/build/lib/remote/blobstore/http/HttpBlobStore.java

```diff
@@ -98,7 +109,44 @@
   @GuardedBy("credentialsLock")
   private long lastRefreshTime;
 
-  public HttpBlobStore(URI uri, int timeoutMillis, @Nullable final Credentials creds)
+
+  public static HttpBlobStore create(URI uri, int timeoutMillis, @Nullable final Credentials creds)
+      throws Exception {
+    return new HttpBlobStore(
+        new InetSocketAddress(uri.getHost(), uri.getPort()),
+        NioEventLoopGroup::new,
+        NioSocketChannel.class,
+        uri, timeoutMillis, creds);
+  }
+
+  public static HttpBlobStore create(
+      DomainSocketAddress domainSocketAddress,
+      URI uri, int timeoutMillis, @Nullable final Credentials creds)
+      throws Exception {
+
+      if (KQueue.isAvailable()) {
+        return new HttpBlobStore(
+            domainSocketAddress,
+            KQueueEventLoopGroup::new,
+            KQueueDomainSocketChannel.class,
+            uri, timeoutMillis, creds);
+      }
+      else if (Epoll.isAvailable()) {
+        return new HttpBlobStore(
+            domainSocketAddress,
+            EpollEventLoopGroup::new,
+            EpollDomainSocketChannel.class,
+            uri, timeoutMillis, creds);
+      }
+      else
+        throw new UnsupportedOperationException("Unix domain sockets are unsupported on this platform");
```

Comment:
This will crash the Bazel server (RuntimeException). Can you throw a checked exception please?